### ISSUE TITLE
Bizarre copy of ARRAY in list assignment at Overload/FileCheck.pm line 578.

### ISSUE THREAD

**Issue by toddr** [CpanelInc/Overload-FileCheck#9]
```perl
#!/usr/bin/perl

use Overload::FileCheck '-from-stat' => \&_mock_stat, q{:check};

my @p;
push @p, bar("/error_log");

exit;

sub bar {
    my ( $path ) = @_;

    open my $fh, '<', $path;

    my $file_size = -f $fh && -s _;
}

sub _mock_stat {
    return [
          64768,
          1095789,
          33188,
          1,
          0,
          0,
          0,
          380,
          1545415173,
          1477585649,
          1477585649,
          4096,
          8
        ];
}
```

**Comment by atoomic** [comment]
63e085b2 is providing a unit test for the stack corruption

**Comment by atoomic** [comment]
this is fixed in v0.007
